client/src/routes/join/+page.svelte

```diff
@@ -0,0 +1,128 @@
+<script lang="ts">
+  import SigninButton from "$components/SigninButton.svelte";
+  import { authStore as user } from "$stores/auth";
+  import { findAndJoinLobby } from "$lib/firebase/join-lobby";
+  import { getUser, createUser, saveDisplayName } from "$lib/firebase/splash";
+  import type { UserData } from "$lib/firebase/firestore-types/users";
+  import { loginAnonymous } from "$lib/firebase/auth";
+  import { goto } from "$app/navigation";
+  import { onMount } from "svelte";
+  import { page } from "$app/stores";
+
+  let name: string = "";
+  let error = {
+    status: false,
+    message: "",
+  };
+  let code: string;
+
+  // Checks to see if the join page has the code query paramter
+  onMount(async () => {
+    console.log(history.state);
+    if ("errorMessage" in history.state) {
+      const { errorMessage } = history.state;
+      error = {
+        status: true,
+        message: errorMessage,
+      };
+      // do this to remove the errorMessage property in state
+      goto("/join", { replaceState: true });
+    }
+    // gets code from url search
+    // the svelte magic with searchparams wasnt working
+    const queryCode = $page.url.search.split("code=")[1];
```

Comment:
this will not work with a url that looks like `/join?code=abcxyz&src=share-button`

why not use something like this?
```ts
$page.url.searchParams.get("code")
```